tor/controller_test.go

```diff
@@ -359,3 +359,33 @@ func TestReconnectSucceed(t *testing.T) {
 	// Check that the connection has been updated.
 	require.NotEqual(t, proxy.clientConn, c.conn)
 }
+
+// TestParseTorReply tests that Tor replies are parsed correctly.
+func TestParseTorReply(t *testing.T) {
+	testCase := []struct {
+		reply          string
+		expectedParams map[string]string
+	}{
+		{
+			// Test a regular reply.
+			reply:          `VERSION Tor="0.4.7.8"`,
+			expectedParams: map[string]string{"Tor": "0.4.7.8"},
+		},
+		{
+			// Test a reply with multiple values, one of them containing spaces.
+			reply:          `AUTH METHODS=COOKIE,SAFECOOKIE,HASHEDPASSWORD COOKIEFILE="/path/with/spaces/Tor Browser/control_auth_cookie"`,
+			expectedParams: map[string]string{"METHODS": "COOKIE,SAFECOOKIE,HASHEDPASSWORD", "COOKIEFILE": "/path/with/spaces/Tor Browser/control_auth_cookie"},
+		},
+		{
+			// Test a multiline reply.
+			reply: `ServiceID=id
+OK`,
+			expectedParams: map[string]string{"ServiceID": "id"},
```

Comment:
Should we add some negative test cases as well?

Reply:
Not sure I understand, what do you mean by negative test cases? The function never fails, although it might return an empty map.

Reply:
Yes, I mean some cases where there are values that won't be covered by the regular expression, for example that `AUTH =invalid` results in an empty result map.

Reply:
And perhaps a test for the special case with the backslash?

Reply:
Oh, okay, I'll add them.

Reply:
This was a good idea, turns out my code didn't handle escaping backslashes correctly (a sequence like `\\`). Fixed it now.